apps/jsonrpc2/lib/jsonrpc2.ex

```diff
@@ -38,60 +38,27 @@ defmodule JSONRPC2 do
     ws = Application.get_env(:jsonrpc2, :ws)
 
     ipc_child = get_ipc_child(ipc)
-    http_ws_child = get_http_ws_child(http, ws)
+    http_configuration = WebSocketHTTP.new(http, :web)
+    ws_configuration = WebSocketHTTP.new(ws, :ws)
 
     children =
-      Enum.filter([ipc_child, http_ws_child], fn
-        nil -> false
-        _ -> true
-      end)
+      Enum.concat(
+        ipc_child,
+        WebSocketHTTP.children(http_configuration, ws_configuration, SpecHandler)
+      )
 
     Supervisor.start_link(children,
       strategy: :one_for_one
     )
   end
 
-  @spec get_ipc_child(Keyword.t()) :: Supervisor.child_spec() | nil
+  @spec get_ipc_child(Keyword.t()) :: Supervisor.child_spec() | []
   defp get_ipc_child(enabled: true, path: path) do
     dirname = Path.dirname(path)
     :ok = File.mkdir_p(dirname)
     _ = File.rm(path)
```

Comment:
why are we creating and then deleting this?

Reply:
this way we ensure that the socket isn't used and that connections get dropped